### PR TITLE
CT-1020: Automatically start tutorials

### PR DESCRIPTION
**Pull request by tomasstark**
This PR addresses [CT-1020](https://laterpay.atlassian.net/browse/CT-1020).

When merchant already went through the tutorial but they have zero contributions active / stored in the database, it launches tutorial automatically next time they go to Contribution builder.

### CODE REVIEW

File: v1/assets/src/js/pages/revenue-generator-contribution.js [hunk 1]
```diff
@@ -113,9 +113,11 @@ window.updateTourProgress = () => {
...
 				if (
 					0 ===
-					parseInt(
-						options.globalOptions.is_contribution_tutorial_completed
-					)
+						parseInt(
+							options.globalOptions
+								.is_contribution_tutorial_completed
+						) ||
+					0 === parseInt( options.has_contributions )
```
Comment (tomasstark) [inline v1/assets/src/js/pages/revenue-generator-contribution.js:120]: Prettier made me do it. 😭

File: v1/assets/src/js/pages/revenue-generator-paywall.js [hunk 2]
```diff
@@ -214,15 +214,19 @@ import { RevGenModal } from '../utils/rev-gen-modal';
...
+							allPurchaseOptions.length > 0 ) ||
 						0 ===
 							parseInt(
-								revenueGeneratorGlobalOptions.globalOptions
-									.paywall_tutorial_done
-							) &&
-						allPurchaseOptions &&
-						allPurchaseOptions.length > 0
+								revenueGeneratorGlobalOptions.has_paywalls
+							)
```
Comment (tomasstark) [inline v1/assets/src/js/pages/revenue-generator-paywall.js:229]: Most of this is prettier again. I added `|| 0 === parseInt( revenueGeneratorGlobalOptions.has_paywalls )` to the stack.

File: v1/inc/classes/class-admin.php [hunk 3]
```diff
@@ -225,6 +225,8 @@ public static function load_assets() {
 				'isPaywallWelcomeDone' => ! empty( $current_global_options['average_post_publish_count'] ),
 			],
 			'has_contributions_only' => $has_contributions_only,
+			'has_contributions'      => ( Contribution::has_contributions() ) ? 1 : 0,
+			'has_paywalls'           => ( Paywall::has_paywalls() ) ? 1 : 0,
```
Comment (tomasstark) [inline v1/inc/classes/class-admin.php:229]: Those are values that will be passed to rendered page as JS object.

File: v2/inc/classes/post-types/class-contribution.php [hunk 4]
```diff
@@ -564,4 +564,26 @@ public static function get_dashboard_url() {
 		return $url;
 	}
 
+	/**
+	 * Check if there are contributions in the database.
+	 *
+	 * @return boolean
+	 */
+	public static function has_contributions() {
```
Comment (tomasstark) [inline v2/inc/classes/post-types/class-contribution.php:572]: May appear as a duplicate (and it is 😄) but the one above is used for Connector integration while this one is for integrating with new platform.

File: v1/assets/src/js/pages/revenue-generator-contribution.js (hunk 1 above)
Comment (leesh22) [reply v1/assets/src/js/pages/revenue-generator-contribution.js:120]: 😩

**Review by leesh22 — APPROVED** [review]

**Pull request merged**